### ISSUE TITLE
Can't download

### ISSUE THREAD

**Issue by BulldozerPete** [meeb/tubesync#389]
I am getting this error: 

![screenshot](https://github.com/meeb/tubesync/assets/61943052/ff4ddffe-b4ba-460d-a143-6a82a1ee6917)

**Comment by meeb** [comment]
That's because `@NintentoAmerica` isn't a channel name. The actual channel ID for `@NintendoAmerica` is `UCGIY_O-8vW4rfX98KlMkvRg` so go to add a source by channel ID and enter `UCGIY_O-8vW4rfX98KlMkvRg`. For example:

https://www.youtube.com/channel/UCGIY_O-8vW4rfX98KlMkvRg/videos

Also see #243, #299, #366 etc.

You can use this tool to find a channels internal YouTube ID: https://www.streamweasels.com/tools/youtube-channel-id-and-user-id-convertor/

**Comment by BulldozerPete** [comment]
Now I have added https://www.youtube.com/channel/UCGIY_O-8vW4rfX98KlMkvRg/videos.

No errors but also no downloads.

**Comment by meeb** [comment]
It's a massive channel, check the tasks tab to see what it's doing. It probably has about 20000 tasks queued up.

**Comment by BulldozerPete** [comment]
Okay.

The channelID is very inconvenient. I have used the option "add a YouTube channel" without channel ID.

Is there also an option to download only new videos?

**Comment by meeb** [comment]
The channel ID stuff is how YouTube works, nothing to do with me. Some `@channels` are not actually channels but custom user pages of mixed playlists as discussed at length in the duplicate issues I linked.

You can adjust your source to only download new videos by setting the "download cap" option on your source settings. Note tubesync still needs to get the metadata for every single video in a channel to get the upload date to know which videos it should download, so regardless of setting a download cap you'll still have a load of tasks for each video to crawl the video metadata when you add a large channel.

Also, you're welcome.

**Comment by BulldozerPete** [comment]
No I can't delete the channel. 

"500 - Internal Server Error

Your request caused an internal server error. This has been logged and our developers will implement a fix shortly."

**Comment by meeb** [comment]
Probably because it's busy indexing a zillion videos and the web interface times out. You can delete it on the command line as well.

```bash
$ docker exec -ti tubesync python3 /app/manage.py delete-source --source uuid-here
```

uuid-here is the UUID of your source, see URL bar when you view a source for the UUID.

**Comment by BulldozerPete** [comment]
> The channel ID stuff is how YouTube works, nothing to do with me. Some `@channels` are not actually channels but custom user pages of mixed playlists as discussed at length in the duplicate issues I linked.
> 
> You can adjust your source to only download new videos by setting the "download cap" option on your source settings. Note tubesync still needs to get the metadata for every single video in a channel to get the upload date to know which videos it should download, so regardless of setting a download cap you'll still have a load of tasks for each video to crawl the video metadata when you add a large channel.
> 
> Also, you're welcome.

I have set the cap to 1 week. it is still queuing up videos,

**Comment by meeb** [comment]
Yes that's because as I said earlier you still need to get the metadata for each video, one at a time. It won't download the videos older than a week but it still needs to check each one.
